### PR TITLE
calculate object address and token address

### PR DESCRIPTION
**Pull request by 0xmaayan**
### Description
Following https://aptos-org.slack.com/archives/C05NLAKJM9U/p1711053989343899 , adding local object address calculation

### CODE REVIEW

**Review by JohnChangUK — APPROVED** [review]
Looks great, verified!

File: tests/unit/address.test.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+describe("address", () => {
+  // Tested against https://explorer.aptoslabs.com/txn/0xd69b71af5333cfb7b90341bd870aa391c76a38c0abce9a293124050556d725b6/changes?network=devnet
+  test("create an object address from creator address and seed", () => {
+    const creatorAddress = "0xfc4974f0336139419b86ab2dc816a8803fe7897c64ecd24e5873690fe66c6986";
+    const seed = "APTOGOTCHI";
+    const address = createObjectAddressFromCreatorAddressAndSeed(creatorAddress, seed);
+    expect(address).toEqual("0x55c4f7cac00e1494cf4c3e079665225b31859fe380dd9e38367b84656faa6e2f");
+  });
+
+  // Tested against https://explorer.aptoslabs.com/object/0x4034bbda2cb1bb0c379a699ac6663b528f4f08b74f36789f367457374fd033a4/resources?network=devnet
```
[the remark this answers is not in the document]
Comment (0xmaayan) [reply tests/unit/address.test.ts:12]: @BriungRi do you have a reference for token object on testnet also?

File: src/core/account/utils/address.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ */
+export const createObjectAddress = (creatorAddress: AccountAddress, seed: Uint8Array | string): string => {
+  const creatorBytes = creatorAddress.bcsToBytes();
+
+  const seedBytes = typeof seed === "string" ? Buffer.from(seed, "utf8") : seed;
+
+  const bytes = new Uint8Array([...creatorBytes, ...seedBytes, DeriveScheme.DeriveObjectAddressFromSeed]);
+
+  return new AccountAddress(sha3_256(bytes)).toString();
+};
```
Comment (gregnazario) [inline src/core/account/utils/address.ts:21]: Why not just return the AccountAddress instead of the string?

Comment (0xmaayan) [reply src/core/account/utils/address.ts:21]: honestly, no strong opinion here - this should be based on end user usage and convenience. Might be better to just return the  AccountAddress and let the end user to figure our what format they want it in

Comment (0xmaayan) [reply src/core/account/utils/address.ts]: updated

**Review by gregnazario — APPROVED** [review]

**Pull request merged**